recursion/recurse.rb

```diff
@@ -1,10 +1,32 @@
 def fact(n)
+  if n == 1 || n == 0
+    return 1
+  end
+  return n * fact(n-1)
 end
 
 def fib(n)
+  if n == 1 || n == 2
+    return 1
+  end
+  return fib(n - 1) + fib(n - 2)
 end
 
 def pal(s)
+  if s.length == 1 || s.length == 0
+    return true
+  end
+
+  first = s[0]
+  last = s[-1]
+
+  if first == last
+    s = s.delete(first)
+    s = s.delete(last)
+    return pal(s)
```

Comment:
Hmm... I'd have to look up efficiency of .delete, but my guess is that it's more expensive than necessary. Instead, we could just pass a smaller string to the recursive call. 

'''return pal( s[1..-2])'''